buildtools/npm/npm.go

```diff
@@ -54,3 +66,23 @@ func (n *NPM) Install(dir string) error {
 	}
 	return nil
 }
+
+func New(options map[string]interface{}) (NPM, error) {
+	npmCmd, _, npmErr := exec.Which("-v", os.Getenv("FOSSA_NPM_CMD"), "npm")
+
+	if npmErr != nil {
+		log.Warnf("Could not find NPM: %s", npmErr.Error())
+	}
+
+	var decodedOptions Options
+	err := mapstructure.Decode(options, &decodedOptions)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return SystemNPM{
+		Cmd:      npmCmd,
+		AllowErr: decodedOptions.AllowNPMErr || true,
```

Comment:
I think we can always ignore errors for now.

We should probably translate NPM exit errors (i.e. errors where the build data is still available) into a special sentinel error interface. If that doesn't make any sense, add a `TODO` and I'll explain tomorrow.